src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocketOptions.cs

```diff
@@ -143,6 +160,52 @@ public TimeSpan KeepAliveInterval
             }
         }
 
+        internal int ReceiveBufferSize => _receiveBufferSize;
+        internal int SendBufferSize => _sendBufferSize;
+        internal ArraySegment<byte>? Buffer => _buffer;
+
+        public void SetBuffer(int receiveBufferSize, int sendBufferSize)
+        {
+            ThrowIfReadOnly();
```

Comment:
@stephentoub What is the design planned for the internal buffers with respect to the WinHTTP implementation? (For the WinHTTP implementation, having an internal buffer would add overheads+extra copying.)